vendors/infineon/boards/xmc4800_plus_optiga_trust_x/aws_tests/application_code/main.c

```diff
@@ -0,0 +1,305 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#include <string.h>
+
+#include "console_io.h"
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Test includes */
+#include "aws_test_runner.h"
+
+/* AWS library includes. */
+#include "iot_system_init.h"
+#include "iot_logging_task.h"
+#include "iot_wifi.h"
+#include "aws_clientcredential.h"
+#include "aws_dev_mode_key_provisioning.h"
+
+/* Logging Task Defines. */
+#define mainLOGGING_MESSAGE_QUEUE_LENGTH    ( 15 )
+#define mainLOGGING_TASK_STACK_SIZE         ( configMINIMAL_STACK_SIZE * 8 )
+
+/* Unit test defines. */
+#define mainTEST_RUNNER_TASK_STACK_SIZE     ( configMINIMAL_STACK_SIZE * 16 )
+
+/* The task delay for allowing the lower priority logging task to print out Wi-Fi 
+ * failure status before blocking indefinitely. */
+#define mainLOGGING_WIFI_STATUS_DELAY       pdMS_TO_TICKS( 1000 )
+
+/* The name of the devices for xApplicationDNSQueryHook. */
+#define mainDEVICE_NICK_NAME                "XMC4800_IOTKIT_TEST"
+
+/**
+ * @brief Application task startup hook for applications using Wi-Fi. If you are not 
+ * using Wi-Fi, then start network dependent applications in the vApplicationIPNetorkEventHook
+ * function. If you are not using Wi-Fi, this hook can be disabled by setting 
+ * configUSE_DAEMON_TASK_STARTUP_HOOK to 0.
+ */
+void vApplicationDaemonTaskStartupHook( void );
+
+/**
+ * @brief Connects to Wi-Fi.
+ */
+static void prvWifiConnect( void );
+
+/**
+ * @brief Initializes the board.
+ */
+static void prvMiscInitialization( void );
+
+/* Ensure the FreeRTOS heap is not crossing SRAM boundaries */
+__attribute__((section("ETH_RAM"))) uint8_t ucHeap[ configTOTAL_HEAP_SIZE ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Application runtime entry point.
+ */
+int main( void )
+{
+    /* Perform any hardware initialization that does not require the RTOS to be
+     * running.  */
+    prvMiscInitialization();
+
+    /* Create tasks that are not dependent on the Wi-Fi being initialized. */
+    xLoggingTaskInitialize( mainLOGGING_TASK_STACK_SIZE,
+                            tskIDLE_PRIORITY,
+                            mainLOGGING_MESSAGE_QUEUE_LENGTH );
+
+    /* FIX ME: If you are using Ethernet network connections and the FreeRTOS+TCP stack,
```

Comment:
Given my understanding that this board is Wi-Fi only, these comments and code blocks could be removed.

Reply:
Agree. Just to explain, we use the code as well with an internal example over the Ethernet, I guess this line originally came from this activity.